packages/utils/src/locking.js

```diff
@@ -0,0 +1,65 @@
+import path from 'path'
+import consola from 'consola'
+import hash from 'hash-sum'
+import fs from 'fs-extra'
+import properlock from 'proper-lockfile'
+import onExit from 'signal-exit'
+
+export const lockPaths = new Set()
+
+export const defaultLockOptions = {
+  stale: 30000,
+  onCompromised: err => consola.fatal(err)
+}
+
+export function getLockOptions(options) {
+  return Object.assign({}, defaultLockOptions, options)
+}
+
+export function createLockPath({ id = 'nuxt', dir, root }) {
```

Comment:
I added the `id` due to a remark of pi0 in the rfc about having separate locks for separate commands. I agree that that wouldnt be very useful, but having the possibility to have different locks on the same dir could be useful. Also the `id` is printed in the error when a lock already exists and could be useful to have a clearer understanding of why the lock exists in the first place